index.js

```diff
@@ -1,29 +1,96 @@
 const path = require('path');
-const fs = require('fs');
+const fs = require('fs').promises;
 const elmCompiler = require('node-elm-compiler');
-const cmdExists = require('command-exists').sync;
+const commandExists = require('command-exists');
 
 const namespace = 'elm';
 const fileFilter = /\.elm$/;
 
-const fileExists = p => fs.existsSync(p) && fs.statSync(p).isFile();
+const fileExists = filePath => fs.stat(filePath).then(stat => stat.isFile()).catch(() => false);
 
-const getPathToElm = () => {
-  if (fileExists('./node_modules/.bin/elm')) return './node_modules/.bin/elm'
-  if (cmdExists('elm')) return 'elm'
+const getPathToElm = async () => {
```

Comment:
If [commandExists rejects with an error](https://www.npmjs.com/package/command-exists#promise), this will break which is why we initially went with the sync api. It's a pretty weird api because sync, returns a boolean but async returns the command itself. Also, it seems like commandExists also works for local files so maybe we can get rid of `fileExists` altogether.

So, we can go with something like this instead...

```js
const getPathToElm = () => {
  const commands = ['./node_modules/.bin/elm', 'elm']
  const CMD_NOT_FOUND_ERR = 'Could not find `elm` executable. You can install it with `yarn add elm` or `npm install elm`'

  return commands
    .reduce(
      (acc, cmd) => acc.catch(_ => commandExists(cmd)),
      Promise.reject(null)
    )
    .catch(_ => Promise.reject(new Error(CMD_NOT_FOUND_ERR)))
}
```

Reply:
I've (hopefully) fixed this, let me know what you think! I've also changed things so that we cache on file modification date rather than content (which should be a lot quicker).